app/models/institution_builder.rb

```diff
@@ -776,6 +777,92 @@ def self.build_versioned_school_certifying_official(version_id)
     SchoolCertifyingOfficial.connection.execute(sql)
   end
 
+  def self.build_institution_category_ratings_for_category(version_id, category)
+    sql = <<-SQL
+      INSERT INTO institution_category_ratings (
+        institution_id,
+        category_name,
+        rated1_count,
+        rated2_count,
+        rated3_count,
+        rated4_count,
+        rated5_count,
+        na_count,
+        average_rating,
+        total_count
+      )
+      SELECT
+        institutions.id,
+        '#{category}',
+        SUM(CASE #{category} WHEN 1 THEN 1 ELSE 0 END),
+        SUM(CASE #{category} WHEN 2 THEN 1 ELSE 0 END),
+        SUM(CASE #{category} WHEN 3 THEN 1 ELSE 0 END),
+        SUM(CASE #{category} WHEN 4 THEN 1 ELSE 0 END),
+        SUM(CASE #{category} WHEN 5 THEN 1 ELSE 0 END),
+        SUM(CASE WHEN #{category} IS NULL THEN 1 ELSE 0 END),
+        (SUM(CASE #{category} WHEN 1 THEN 1 ELSE 0 END)
+         + SUM(CASE #{category} WHEN 2 THEN 2 ELSE 0 END)
+         + SUM(CASE #{category} WHEN 3 THEN 3 ELSE 0 END)
+         + SUM(CASE #{category} WHEN 4 THEN 4 ELSE 0 END)
+         + SUM(CASE #{category} WHEN 5 THEN 5 ELSE 0 END)) / COUNT(institutions.id)::float,
```

Comment:
Could we use the alias' here like SUM(rated1_count + rated2_count * 2 + rated3_count * 3 ...?

Reply:
Just tested to see if that'd work and it looks like it isn't defined yet at that point.

Reply:
Makes sense, thanks